Services/Notifications/classes/ilNotificationOSDRepository.php

```diff
@@ -0,0 +1,117 @@
+<?php declare(strict_types=1);
+
+namespace ILIAS\Notifications;
+
+use ilDBInterface;
+use ILIAS\DI\Container;
+use ILIAS\Notifications\Model\ilNotificationConfig;
+use ILIAS\Notifications\Model\ilNotificationLink;
+use ILIAS\Notifications\Model\ilNotificationObject;
+use ilLanguage;
+
+/******************************************************************************
+ *
+ * This file is part of ILIAS, a powerful learning management system.
+ *
+ * ILIAS is licensed with the GPL-3.0, you should have received a copy
+ * of said license along with the source code.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ *      https://www.ilias.de
+ *      https://github.com/ILIAS-eLearning
+ *
+ *****************************************************************************/
+
+/**
+ * @author Jan Posselt <jposselt@databay.de>
+ */
+class ilNotificationOSDRepository
+{
+    private static ?self $instance = null;
+    private ilDBInterface $database;
+
+    public static function getInstance() : self
```

Comment:
Do we really need to introduce a singleton here?

Maybe you can add something like `NotificationSercies` to the `$DIC` and add a method there to retrieve a repository?

Reply:
I Agree, but would prefer todo so in a later PR since this dont seem to be relevant to be PHP8 usable and is more of a general refactoring.